content/en/hosting/4.x/upgrade-troubleshooting.md

```diff
@@ -0,0 +1,153 @@
+weight: 50
+aliases:
+  - 
+description: >
+  What to do when CHT 4.x upgrades don't work as planned
+relatedContent: >
+  hosting/4.x/data-migration
+---
+
+4.0.0 was released in November of 2022 so 4.x is mature and users have learned a number of important lessons on how to fix failed 4.x upgrades.  Below are some specific tips as well as general practices on upgrading 4.x.
+
+There's a concept of upgrades "getting stuck" which mainly means that after many many hours an upgrade is not making any progress.  Most likely, this will manifest as the progress bars in the  upgrade admin web UI not increasing and "sticking" at a certain percentage.
+
+{{% pageinfo %}}
+All tips apply to both [Docker]({{< relref "hosting/4.x/production/docker" >}}) and [Kubernetes]({{< relref "hosting/4.x/production/kubernetes" >}}) based deployments unless otherwise specified.
+
+All upgrades are expected to succeed without issue.  Do not attempt any fixes unless you actively have a problem upgrading.
+{{% /pageinfo %}}
+
+## Considerations
+
+When troubleshooting, consider making sure there are:
+
+* Backups exist and restores have been tested 
+* Extra disk space is availabe (up to 5x!)
+* The upgrade has been tested on a development instance with production data
+
+## A go-to fix: restart
+
+A safe fix for any upgrade getting stuck is to restart all services.  Any views that were being re-indexed will be picked up where they left off without losing any work.  This should be your first step when trouble shooting a stuck upgrade. 
+
+If you're able to, after a restart go back into the admin web GUI and try to upgrade again.  Consider trying this at least twice.
+
+## CHT 4.0.x - 4.3.x: CouchDB Crashes
+
+**[Issue #9286](https://github.com/medic/cht-core/issues/9286)**:  Starting an upgrade that involves view indexing can cause CouchDB to crash on large databases (>30m docs).  The upgrade will fail and you will see the logs below when you have this issue.
+
+HAProxy:
+
+```shell
+<150>Jul 26 20:57:39 haproxy[12]: 172.22.0.4,<NOSRV>,503,0,0,0,GET,/,-,medic,'-',217,-1,-,'-'
+<150>Jul 26 20:57:40 haproxy[12]: 172.22.0.4,<NOSRV>,503,0,0,0,GET,/,-,medic,'-',217,-1,-,'-'
+<150>Jul 26 20:57:41 haproxy[12]: 172.22.0.4,<NOSRV>,503,0,0,0,GET,/,-,medic,'-',217,-1,-,'-'
+```
+
+CouchDB
+
+```shell
+[notice] 2024-07-26T20:52:45.229027Z couchdb@127.0.0.1 <0.10998.4> dca1387a05 haproxy:5984 172.22.0.4 medic GET /_active_tasks 200 ok 1
+[notice] 2024-07-26T20:52:45.234397Z couchdb@127.0.0.1 <0.10986.4> 2715cd1e47 haproxy:5984 172.22.0.4 medic GET /medic-logs/_all_docs?descending=true&include_docs=true&startkey=%22upgrade_log%3A1722027165223%3A%22&limit=1 200 ok 6
+[notice] 2024-07-26T20:52:45.468469Z couchdb@127.0.0.1 <0.11029.4> 3bf85c6071 haproxy:5984 172.22.0.4 medic GET /_active_tasks 200 ok 1
+```
+
+**Fix:**
+1. Check that all the indexes are warmed by loading them one by one in fauxton.
+2. Restart all services, **retry** upgrade from Admin GUI - do not cancel and upgrade.
```

Comment:
Maybe we can include an image of the retry button to be super clear

Reply:
yes! i added on in f0ae84fc . I wasn't sure exactly how to recreate that locally, so it's a real screenshot, but slightly doctored version strings to meet our use case.  Lemme know if it's not accurate!

see f0ae84fc